unsafe/src/test/java/org/apache/spark/unsafe/types/IntervalSuite.java

```diff
@@ -56,4 +56,34 @@ public void toStringTest() {
     i = new Interval(34, 3 * MICROS_PER_WEEK + 13 * MICROS_PER_HOUR + 123);
     assertEquals(i.toString(), "interval 2 years 10 months 3 weeks 13 hours 123 microseconds");
   }
+
+  @Test
+  public void fromStringTest() {
+    String s;
+    Interval i;
+
+    s = "interval  2 weeks -6 minute";
```

Comment:
add a unit test where there is only # month specified, but nothing else.
and another one with only # year.
and then one more for year + month.

and one for only day, week, ... all the units.